src/packages/auth/src/errors.ts

```diff
@@ -23,3 +26,63 @@ export class ChallengeError extends ApolloError {
 		Object.defineProperty(this, 'name', { value: 'ChallengeError' });
 	}
 }
+
+enum RestrictedFieldErrorCode {
+	GRAPHQL_VALIDATION_FAILED = 'GRAPHQL_VALIDATION_FAILED',
+	BAD_USER_INPUT = 'BAD_USER_INPUT',
+}
+
+export enum FieldLocation {
+	FIELD = 'FIELD',
+	FILTER = 'FILTER',
+	NESTED_FILTER = 'NESTED_FILTER',
+	INPUT = 'INPUT',
+}
+
+export class RestrictedFieldError extends ApolloError {
+	constructor(
+		private entityName: string,
+		private field: FieldDetails
+	) {
+		super('');
+
+		switch (this.field.location) {
+			case FieldLocation.FIELD:
+				this.formatFieldMessage();
+				break;
+			case FieldLocation.NESTED_FILTER:
+				this.formatFilterMessage();
+				break;
+			case FieldLocation.FILTER:
+				this.formatArgMessage();
+				break;
+			case FieldLocation.INPUT:
+				this.formatArgMessage();
+				break;
```

Comment:
Do we need a `default:` here, or is the default already good enough?

Reply:
🤔 Should we error maybe if it does not match?